### PR TITLE
Develop transit

### LINKED ISSUE #35: [FEATURE] Add Transit Assignment
### User Story
*As a model user I'd like to be able to run transit assignment and skims within `tm2py` *

- [ ] User story satisfied
- [ ] Doc strings
- [ ] General documentation
- [ ] Tests
- [ ] Passing tests

### Level of Effort
Medium

### Who should be involved?
Implementer: 
Commenters: 
Users: 
Reviewers: 

### Risk
Medium

#### Tests
*What are relevant tests or what tests need to be created in order to determine that this issue is complete?*

- [ ] Add assignment test
- [ ] Add skimming test
- [ ] other?

### PR DESCRIPTION
**Pull request by inrokevin**
## What existing problem does the pull request solve and why should we include it?
Adding updated transit assignment and skims implementation.

Basic (essential) documentation with typehints in place, additional documentation to follow by MTC / future task order.

## Testing plan
 - [  ] Code for this PR is covered in tests 
 - [X] Basic "does it run" test added, more testing to follow by MTC / future task order
 - [x] Code passes all existing tests
 Note that the mkdocs is not working on develop so doesn't pass on this branch either.
 - [X] Code linted

Closes #35

### CODE REVIEW

File: tm2py/components/component.py [hunk 1]
```diff
@@ -62,31 +90,23 @@ def controller(self):
         """Parent controller."""
         return self._controller
 
+    @property
+    def emme_manager(self):
+        return self.controller.emme_manager
+
```
Comment (e-lo) [inline tm2py/components/component.py:96]: Adding this here so we don't have to have it everywhere this method is implemented.

File: tm2py/components/network/transit/transit_assign.py [hunk 2]
```diff
@@ -1,9 +1,632 @@
...
+        duration_lookup = dict(
+            (p.name, p.length_hours) for p in self.config.time_periods
+        )
+        return duration_lookup[self._time_period]
+
+    def _run_ccr_assign(self):
+        transit_classes = self._transit_classes
+        assign_transit = self.controller.emme_manager.tool(
+            "inro.emme.transit_assignment.capacitated_transit_assignment"
+        )
```
Comment (e-lo) [inline tm2py/components/network/transit/transit_assign.py:211]: Question for INRO: Is the only way to send a cost function as a string? This seems rather nonideal for many reasons.

Comment (e-lo) [reply tm2py/components/network/transit/transit_assign.py:211]: (ditto for headway cost function)

File: tm2py/components/network/transit/transit_assign.py [hunk 3]
```diff
@@ -1,9 +1,632 @@
...
+        code = compile(
+            _HEADWAY_COST_FUNCTION.format(mode_name),
+            "headway_cost_function",
+            "exec",
+        )
+        # Yes pylint, I know exec is being used here
+        # pylint: disable=W0122
+        exec(code, enclosing_scope)
+        calc_eawt = enclosing_scope["calc_eawt"]
+        hdwy_fraction = 0.5  # fixed in assignment spec
```
Comment (e-lo) [inline tm2py/components/network/transit/transit_assign.py:395]: what is EAWT?

File: tm2py/components/network/transit/transit_assign.py [hunk 4]
```diff
@@ -1,9 +1,632 @@
...
+            msg = f"Transit assignment for period {self._time_period}"
+            with self.logger.log_start_end(msg):
+                self._scenario = self.get_emme_scenario(emmebank, self._time_period)
+                if use_ccr:
+                    self._run_ccr_assign()
+                    self._calc_segment_ccr_penalties()
+                else:
+                    self._run_extended_assign()
+                if self.config.transit.output_stop_usage_path is not None:
+                    network, class_stop_attrs = self._calc_connector_flows()
```
Comment (e-lo) [inline tm2py/components/network/transit/transit_assign.py:162]: do we want to calculate the connector flows to the network even if we don't export them?

File: tm2py/components/network/transit/transit_assign.py [hunk 5]
```diff
@@ -1,9 +1,632 @@
...
+
+        Accounts for fare table (mapping from input mode ID to auto-generated
+        set of mode IDs for fare transition table (fares.far input) by applyfares
+        component.
+        """
+        if self._transit_config.use_fares:
+            out_modes = set([])
+            for mode in modes:
+                if mode.assign_type == "TRANSIT":
+                    out_modes.update(self._fare_modes[mode.mode_id])
```
Comment (e-lo) [inline tm2py/components/network/transit/transit_assign.py:551]: the assignment of something to be used in fare calculation shiould be done separately - not hidden in this method

**Comment by i-am-sijia** [comment]
All tests passing locally with Emme, except for make docs and household model.

**Pull request merged**